src/formatter.ts

```diff
@@ -560,12 +575,60 @@ export class Formatter {
   }
 
   /** Create a `ModifierContext` for each tick in `voices`. */
-  createModifierContexts(voices: Voice[]): AlignmentContexts<ModifierContext> {
-    const fn: addToContextFn<ModifierContext> = (tickable: Tickable, context: ModifierContext) =>
-      tickable.addToModifierContext(context);
-    const contexts = createContexts(voices, () => new ModifierContext(), fn);
-    this.modifierContexts = contexts;
-    return contexts;
+  createModifierContexts(voices: Voice[]) {
+    if (voices.length == 0) return;
+    const resolutionMultiplier = Formatter.getResolutionMultiplier(voices);
+
+    // Initialize tick maps.
+    const tickToContextMap: Map<Stave | undefined, Record<number, ModifierContext>> = new Map();
```

Comment:
The issue as I understand it is that 2 tickables at the same tickcontext can have staves at different y values.  So the MC should be different so the y offset is calculated correctly if it is a staff offset vs. a notehead offset.  But it's also true that stave does not have to be defined here, so we are mapping in Stave | undefined.  That seems kinda gross :)   Can we map instead on the y value of the staves?  If a tickable does not have a stave, we just use 0.  Then it can just be `Record<number, Record <number, ModifierContext>>` .

Reply:
Yes we could do that, but theoretically we can move the Staves with setY making it difficult to know which is the associated Stave. Anyway I am happy to change if this is a prefered approach.

Reply:
Now that I think of it, I don't prefer it.  Indexing by 'y' seems like a side-effect, and like you say, the staves could start out with same y and change later.  

It seems like this could be simpler though. Even the original code seems too complicated...

``` javascript
const postFormatContexts = (contexts: AlignmentContexts<ModifierContext> | AlignmentContexts<TickContext>) =>
      contexts.list.forEach((tick) => contexts.map[tick].postFormat());
```
Can't it just be this?
``` javascript
contexts.array.forEach((mc) => mc.postFormat());
```

We should only need a key from the `staff.attrs.id` and `integerTicks`.  So like 'vf-101-1024'.  If there is no stave it is just '-1024'.  Then we can have just `tickToContextMap: Record<string, ModifierContext>` to keep track of MC we've already created.  And once we've called `tickable.addToModifierContext(context);` for all tickables, we don't care about that relationship anymore.  We just need to call `postFormat` on all the MCs, which we can do from an array.

It's late here, so I may be missing something...

Reply:
@0xfe do you have a preference?

Reply:
@AaronDavidNewman I have simplified this function removing `postFormatContext` in general the template functions using AlignmentContexts<T> are no longer required to be templates and could be re-writen.

Reply:
@AaronDavidNewman i believe that the approach you proposed did not work because the TickContexts have to be formatted in tick order. I just thought that this probably not the case for ModifierContexts, I have to try.

Reply:
For the record, what I said above is incorrect.  We need to call `postFormat` on the MC _in_ _tick_ _order_.  That is the bug here, postFormat was called for all the ticks in one stave, and then all the ticks in a different stave.